pkg/list/list_vendor.go

```diff
@@ -0,0 +1,309 @@
+	var allVendors []schema.AtmosVendorSource
+	for _, f := range files {
+		vendors, err := processVendorFile(f, atmosConfig)
+		if err != nil {
+			return "", err
+		}
+		allVendors = append(allVendors, vendors...)
+	}
+
+	// Convert vendor info to rows based on header columns
+	var rows [][]string
+	for _, vendor := range allVendors {
+		row := make([]string, len(header))
+		for i, col := range header {
+			var colConfig *schema.ListColumnConfig
+			for _, c := range listConfig.Columns {
+				if c.Name == col {
+					colConfig = &c
+					break
+				}
+			}
+
+			// If no column config found, use default mapping
+			if colConfig == nil {
+				switch col {
+				case "Component":
+					row[i] = vendor.Component
+				case "Version":
+					row[i] = vendor.Version
+				case "Folder":
+					if len(vendor.Targets) > 0 {
+						row[i] = filepath.Join(atmosConfig.BasePath, vendor.Targets[0])
+					}
+				case "Source":
+					row[i] = vendor.Source
+				case "Type":
+					row[i] = "Vendor Manifest"
+				case "Manifest":
+					row[i] = vendor.File
+				}
+				continue
+			}
+
+			data := map[string]interface{}{
+				"Component":           vendor.Component,
+				"Version":             vendor.Version,
+				"Type":                "Vendor Manifest",
+				"File":                vendor.File,
+				"Source":              vendor.Source,
+				"Target":              "",
+				"Tags":                vendor.Tags,
+				"atmos_component":     vendor.Component,
+				"atmos_vendor_type":   "Vendor Manifest",
+				"atmos_vendor_file":   vendor.File,
+				"atmos_vendor_target": "",
+			}
+
+			// Add target path if available
+			if len(vendor.Targets) > 0 {
+				target := filepath.Join(atmosConfig.BasePath, vendor.Targets[0])
+				data["Target"] = target
+				data["atmos_vendor_target"] = target
+			}
+
+			// Process the template
+			processed, err := exec.ProcessTmpl(col, colConfig.Value, data, false)
+			if err != nil {
+				row[i] = fmt.Sprintf("Error: %v", err)
+				continue
+			}
+			row[i] = processed
+		}
+		rows = append(rows, row)
+	}
+
+	// Sort rows for consistent output
+	sort.Slice(rows, func(i, j int) bool {
+		// Compare each column in order until we find a difference
+		for col := 0; col < len(rows[i]); col++ {
+			if rows[i][col] != rows[j][col] {
+				return rows[i][col] < rows[j][col]
+			}
+		}
+		return false // rows are identical
+	})
+
+	if len(rows) == 0 {
+		return "No vendor configurations found", nil
+	}
+
+	// Handle different output formats
+	switch format {
+	case FormatJSON:
+		jsonBytes, err := json.MarshalIndent(allVendors, "", "  ")
+		if err != nil {
+			return "", fmt.Errorf("error formatting JSON output: %w", err)
+		}
+		return string(jsonBytes), nil
+
+	case FormatCSV, FormatTSV:
+		var output strings.Builder
+		output.WriteString(strings.Join(header, delimiter) + utils.GetLineEnding())
+		for _, row := range rows {
+			output.WriteString(strings.Join(row, delimiter) + utils.GetLineEnding())
+		}
+		return output.String(), nil
```

Comment:
We should be using the idiomatic way of creating CSV formatted output using `encoding/csv`